src/events/interaction.ts

```diff
@@ -1,48 +1,35 @@
-import { time } from "@discordjs/builders";
 import { Interaction } from "discord.js";
 import { CommandSupport } from "../Command";
-import { parseTime } from "../round";
-import { PersistentTimer } from "../timer";
-import { UserError } from "../util/errors";
+import { TimerCommand } from "../slash/timer";
+import { AutocompletableCommand, SlashCommand } from "../SlashCommand";
+import { serialiseInteraction } from "../util";
 import { getLogger } from "../util/logger";
 
 const logger = getLogger("interaction");
 
+// eslint-disable-next-line @typescript-eslint/explicit-function-return-type
 export function makeHandler({ organiserRole, timeWizard }: CommandSupport) {
-	return async function interactionCreate(interaction: Interaction): Promise<void> {
-		if (!interaction.isChatInputCommand() || !interaction.inCachedGuild() || interaction.commandName !== "timer") {
-			return;
-		}
-		// Should be replaced by the built-in system
-		const role = await organiserRole.get(interaction.guild);
-		if (!interaction.member.roles.cache.has(role)) {
-			logger.verbose(`Rejected /timer attempt from ${interaction.user} in ${interaction.guildId}.`);
-			await interaction.reply({ content: `You cannot use this.`, ephemeral: true });
-			return;
+	const commandArray = [
```

Comment:
I assume this also has to be updated when a new command is added, in addition to the index?

Reply:
The two locations are here (construct object for running code) and src/cli/register-slash.ts (metadata for registering the command with Discord). There are two spots because I didn't pull in tsyringe